great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -24,11 +26,29 @@
 from great_expectations.rule_based_profiler.types.data_assistant_result import (
     DataAssistantResult,
 )
-from great_expectations.util import measure_execution_time
+from great_expectations.util import camel_to_snake, measure_execution_time
 from great_expectations.validator.validator import Validator
 
 
-class DataAssistant(ABC):
+# noinspection PyMethodParameters
+class MetaDataAssistant(ABCMeta):
+    """
+    MetaDataAssistant registers every DataAssistant class as it is defined, it them to the DataAssistant registry.
+
+    Any class inheriting from DataAssistant will be registered by snake-casing the name of the class.
+    """
+
+    def __new__(cls, clsname, bases, attrs):
+        newclass = super().__new__(cls, clsname, bases, attrs)
+        # noinspection PyUnresolvedReferences
+        if not newclass.is_abstract():
+            newclass.data_assistant_type = camel_to_snake(name=clsname)
+            register_data_assistant(data_assistant=newclass)
+
+        return newclass
+
```

Comment:
Please annotate (as best as you can)

Reply:
@cdkini I added a docstring (committed).  Thanks!